jxl/src/api/inner/codestream_parser/sections.rs

```diff
@@ -56,7 +56,9 @@ impl CodestreamParser {
             self.available_sections.push(s);
         }
         if self.available_sections.is_empty() {
-            return Ok(());
+            return Ok(Some(
+                self.sections.front().unwrap().len - self.ready_section_data,
```

Comment:
I believe this should be a saturating_sub.

Reply:
Really?
At this point we know sections isn't empty, and we have no sections with len <= ready section data, right?

Reply:
Yeah, that's what I think, but maybe it pays off to be defensive?

Reply:
I think you're right, let's leave this as is unless it does indeed crash sometimes